crates/subspace-node/src/domain.rs

```diff
@@ -50,3 +55,71 @@ impl NativeExecutionDispatch for EVMDomainExecutorDispatch {
         evm_domain_runtime::native_version()
     }
 }
+
+/// [`DomainGenesisBlockBuilder`] is used on the consensus node for building the
+/// domain genesis block from a specific serialized domain runtime genesis config.
+pub struct DomainGenesisBlockBuilder<Block, B, E> {
+    backend: Arc<B>,
+    executor: E,
+    _phantom: PhantomData<Block>,
+}
+
+impl<Block, B, E> DomainGenesisBlockBuilder<Block, B, E>
+where
+    Block: BlockT,
+    B: Backend<Block>,
+    E: RuntimeVersionOf + Clone,
+{
+    /// Constructs a new instance of [`DomainGenesisBlockBuilder`].
+    pub fn new(backend: Arc<B>, executor: E) -> Self {
+        Self {
+            backend,
+            executor,
+            _phantom: Default::default(),
+        }
+    }
+
+    /// Constructs the genesis domain block from a serialized runtime genesis config.
+    pub fn generate_genesis_block(
+        &self,
+        runtime_type: RuntimeType,
+        raw_runtime_genesis_config: Vec<u8>,
+    ) -> sp_blockchain::Result<Block> {
+        let domain_genesis_block_builder = match runtime_type {
+            RuntimeType::Evm => {
+                let runtime_genesis_config: evm_domain_runtime::RuntimeGenesisConfig =
+                    serde_json::from_slice(&raw_runtime_genesis_config)
+                        .map_err(|err| sp_blockchain::Error::Application(Box::new(err)))?;
```

Comment:
As discussed with @vedhavyas, we may not have `raw_runtime_genesis_config` as input from the runtime caller, instead `RuntimeGenesisConfig` should be an empty value.

Reply:
Okay